src/library/dao/trackdao.cpp

```diff
@@ -1361,38 +1365,30 @@ TrackPointer TrackDAO::getTrackFromDB(TrackId trackId) const {
 
     // If the track is dirty send dirty notifications after we inserted
     // it in the cache. BaseTrackCache cares about dirty notifications
-    // and the setDirty call above happens before we connect to the
+    // and the markDirty() call above happens before we connect to the
     // track's signals.
-    if (shouldDirty) {
+    if (isTrackDirty) {
         emit(trackDirty(trackId));
+    } else {
+        emit(trackClean(trackId));
     }
 
-    // If the header hasn't been parsed, parse it but only after we set the
-    // track clean and hooked it up to the track cache, because this will
-    // dirty it.
-    SoundSourceProxy(pTrack).loadTrackMetadata();
```

Comment:
I do not understand why this call was here. So I am afraid I cannot tell if removing this is a good idea.
Did this always reload the track Metadata from file? Is it a regression from a previous commit? 
It should be possible to have different Metadata in Mixxx and the file. Was this broken before?